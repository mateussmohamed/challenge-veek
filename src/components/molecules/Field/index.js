import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Label, Input } from 'atoms'

const StyledField = styled.div`
  display: block;
  margin-bottom: 25px;
`

const Field = props => {
  return (
    <StyledField {...props}>
      <Label>{props.label}</Label>
      <Input
        id={props.name}
        name={props.name}
        type={props.type}
        onChange={props.onChange}
      />
    </StyledField>
  )
}

Field.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
}

export default Field
