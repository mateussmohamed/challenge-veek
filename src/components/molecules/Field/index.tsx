import React from 'react'
import styled from 'styled-components'

import { Label, File, TextField, TextArea } from '../../atoms'

const StyledField = styled.div`
  display: block;
  margin-bottom: 25px;
`

const ErrorMessage = styled.p`
  color: red;
  font-size: 12px;
  margin: 8px 0 0 0;
`

interface FieldProps {
  label: string
  name: string
  type: string
  value: string
  error: string
  onChange: (e: React.FormEvent<HTMLInputElement>) => void
}

const getComponent = props => ({
  textarea: <TextArea {...props} />,
  text: <TextField {...props} />,
  file: <File {...props} />,
})

export const Field = ({ error, ...props }: FieldProps) => {
  return (
    <StyledField>
      <Label>{props.label}</Label>
      {getComponent(props)[props.type]}
      {error && <ErrorMessage>{error}</ErrorMessage>}
    </StyledField>
  )
}
