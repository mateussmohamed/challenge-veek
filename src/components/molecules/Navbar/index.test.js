import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import Field from './index'

describe('[<Field />]', () => {
  const wrap = (props = {}) => shallow(<Field {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap({
      label: 'First Name',
      name: 'name',
    })

    expect(wrapper).toMatchSnapshot()
  })

  describe('when pass type prop', () => {
    test('matches to snapshot', () => {
      const wrapper = wrap({
        label: 'Upload your Avatar',
        name: 'avatar',
        type: 'file',
      })

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when pass multiline prop', () => {
    test('matches to snapshot with textarea', () => {
      const wrapper = wrap({
        label: 'Comments',
        name: 'comments',
        multiline: true,
      })

      expect(wrapper).toMatchSnapshot()
    })
  })
})
