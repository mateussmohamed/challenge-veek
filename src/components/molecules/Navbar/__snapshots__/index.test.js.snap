// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<Field />] matches to snapshot 1`] = `
<Navbar__NavbarStyled>
  <ToggleMenu
    onClick={[Function]}
  />
  <NavItems
    className=""
  />
</Navbar__NavbarStyled>
`;

exports[`[<Field />] when pass multiline prop matches to snapshot with textarea 1`] = `
<Navbar__NavbarStyled>
  <ToggleMenu
    onClick={[Function]}
  />
  <NavItems
    className=""
  />
</Navbar__NavbarStyled>
`;

exports[`[<Field />] when pass type prop matches to snapshot 1`] = `
<Navbar__NavbarStyled>
  <ToggleMenu
    onClick={[Function]}
  />
  <NavItems
    className=""
  />
</Navbar__NavbarStyled>
`;
